frontend/hooks/useFeatureFlag.ts

```diff
@@ -0,0 +1,50 @@
+import { z } from 'zod';
+
+import { AgentType } from '@/enums/Agent';
+import { assertRequired } from '@/types/Util';
+
+import { useServices } from './useServices';
+
+const FeatureFlagsSchema = z.enum(['last-transactions', 'balance-breakdown']);
+type FeatureFlags = z.infer<typeof FeatureFlagsSchema>;
+
+const FeaturesConfigSchema = z.record(
+  z.nativeEnum(AgentType),
+  z.record(FeatureFlagsSchema, z.boolean()),
+);
+
+const FEATURES_CONFIG = FeaturesConfigSchema.parse({
+  [AgentType.PredictTrader]: {
+    'balance-breakdown': true,
+    'last-transactions': false,
+  },
+});
+
+/**
+ * Hook to check if a feature flag is enabled for the selected agent
+ * @example const isFeatureEnabled = useFeatureFlag('feature-name');
+ */
+export const useFeatureFlag = (featureFlag: FeatureFlags | FeatureFlags[]) => {
+  const { selectedAgentType } = useServices();
+
+  // Ensure an agent is selected before using the feature flag
+  assertRequired(
+    selectedAgentType,
+    'Feature Flag must be used within a ServicesProvider',
+  );
+
+  // Ensure the selected agent type is supported
+  const selectedAgentFeatures = FEATURES_CONFIG[selectedAgentType];
+  assertRequired(
```

Comment:
tbh, I don't think both `assertRequired`s are needed, we don't allow `selectedAgentType` to be null or undefined, and I assume `FEATURES_CONFIG` will forse us to define all types from `AgentType`, so typescript should prevent us from making mistakes here.

in general, the idea is good, but it suits better for js, what do you think?

Reply:
it'd work if we used typescript only, the partial thing is causing the error

<img width="689" alt="image" src="https://github.com/user-attachments/assets/5cf2a92f-dd1e-469a-81a7-3da431fef967">

and if I uncomment e.g. optimus in AgentType, it'll start complaining also
 
<img width="935" alt="image" src="https://github.com/user-attachments/assets/ff723b20-2507-47be-a231-1ad9a1693b0b">

but okay, fine with me